### PR TITLE
fix time snap with interval month

### PR DESCRIPTION
**Pull request by michaelmattig**
https://trello.com/c/8522Cwl2

### CODE REVIEW

**Comment by michaelmattig** [comment]
[GdalSource doesn't yield correct time slices for dates before start](https://trello.com/c/8522Cwl2/211-gdalsource-doesnt-yield-correct-time-slices-for-dates-before-start)

File: datatypes/src/primitives/time_step.rs [hunk 1]
```diff
@@ -905,6 +939,21 @@ mod tests {
...
+    #[test]
+    fn time_snap_0() {
+        let time_snapper = TimeStep {
+            granularity: TimeGranularity::Months,
+            step: 1,
+        };
+
+        assert_eq!(
+            time_snapper
+                .snap_relative(1_388_534_400_000.into(), 0.into())
```
Comment (jdroenner) [inline datatypes/src/primitives/time_step.rs:951]: could you add a comment what this number represents?

File: operators/src/source/gdal_source.rs [hunk 2]
```diff
@@ -1530,4 +1532,44 @@ mod tests {
...
+                ),
+            },
+            &query_ctx,
+        );
+        let c: Vec<Result<RasterTile2D<u8>>> = q.collect().await;
+
+        assert_eq!(c.len(), 1);
+        assert_eq!(
+            c[0].as_ref().unwrap().time,
+            TimeInterval::new_unchecked(0, 2_678_400_000)
```
Comment (jdroenner) [inline operators/src/source/gdal_source.rs:1572]: could you add a comment what this number represents?

**Review by jdroenner — CHANGES_REQUESTED** [review]
please add comments to explain magic numbers and check if we can use rem_euclid

File: datatypes/src/primitives/time_step.rs [hunk 3]
```diff
@@ -34,6 +34,15 @@ pub struct TimeStep {
     pub step: u32,
 }
 
+/// mathematical modulo that correctly handles negative `a`s
+fn modulo(a: i32, n: i32) -> i32 {
```
[the remark this answers is not in the document]
Comment (michaelmattig) [reply datatypes/src/primitives/time_step.rs]: you are right, I was somehow convinced that it's something differnent, but I checked again and it works. 3300ec0b476320d9e6705ea58230fe3647ce4283

File: datatypes/src/primitives/time_step.rs (hunk 1 above)
Comment (michaelmattig) [reply datatypes/src/primitives/time_step.rs:943]: 3300ec0b476320d9e6705ea58230fe3647ce4283

File: operators/src/source/gdal_source.rs (hunk 2 above)
Comment (michaelmattig) [reply operators/src/source/gdal_source.rs]: 3300ec0b476320d9e6705ea58230fe3647ce4283

**Pull request merged**